azurelinuxagent/ga/update.py

```diff
@@ -148,26 +135,49 @@ def run_latest(self):
                     agent_cmd,
                     ret)
                 logger.warn(msg)
-                add_event(u"WALA", is_success=False, message=msg)
+                add_event(
+                    AGENT_NAME,
+                    version=agent_version,
+                    op=WALAEventOperation.Enable,
+                    is_success=False,
+                    message=msg)
                 if latest_agent is not None:
                     latest_agent.mark_failure()
             else:
                 msg = u"Agent {0} launched with command '{1}' returned 0".format(
                     agent_name,
                     agent_cmd)
                 logger.info(msg)
-                add_event(u"WALA", is_success=True, message=msg)
+                add_event(
+                    AGENT_NAME,
+                    version=agent_version,
+                    op=WALAEventOperation.Enable,
+                    is_success=True,
+                    message=msg)
         except Exception as e:
             msg = u"Agent {0} launch failed with command '{1}' failed with exception: {2}".format(
                 agent_name,
                 agent_cmd,
                 ustr(e))
             logger.warn(msg)
-            add_event(u"WALA", is_success=False, message=msg)
+            add_event(
```

Comment:
@boumenot Considered and rejected for now. It should be done across the codebase and not just in this file. That change was too large, so I continued the pattern used elsewhere.